### PR TITLE
Upgrade mobile app to angular 13

### PR DESCRIPTION
**Pull request by yksolanki9**

### CODE REVIEW

**Review by Dimple16 — APPROVED** [review]
only method placement changes included with angular upgrade, not checking method codes

File: src/app/fyle/add-edit-mileage/fy-select-vehicle/fy-select-vehicle.component.ts [hunk 1]
```diff
@@ -27,22 +27,18 @@ export class FySelectVehicleComponent implements OnInit, ControlValueAccessor {
...
-
   private onTouchedCallback: () => void = noop;
 
   private onChangeCallback: (_: any) => void = noop;
 
   constructor(private injector: Injector) {}
 
-  ngOnInit() {
-    this.ngControl = this.injector.get(NgControl);
+  get valid() {
```
Comment (rvab) [inline src/app/fyle/add-edit-mileage/fy-select-vehicle/fy-select-vehicle.component.ts:36]: return `data type`?

File: src/app/shared/components/fy-add-to-report/fy-add-to-report.component.ts [hunk 2]
```diff
@@ -99,6 +93,12 @@ export class FyAddToReportComponent implements OnInit, OnDestroy {
     }
   }
 
+  ngOnInit() {
+    this.ngControl = this.injector.get(NgControl);
+  }
+
+  ngOnDestroy(): void {}
```
Comment (rvab) [inline src/app/shared/components/fy-add-to-report/fy-add-to-report.component.ts:100]: Is it needed?

File: src/app/shared/components/fy-select/fy-select.component.ts [hunk 3]
```diff
@@ -115,6 +109,12 @@ export class FySelectComponent implements ControlValueAccessor, OnInit, OnDestro
     }
   }
 
+  ngOnInit() {
+    this.ngControl = this.injector.get(NgControl);
+  }
+
+  ngOnDestroy(): void {}
```
Comment (rvab) [inline src/app/shared/components/fy-select/fy-select.component.ts:116]: is it needed?

**Review by rvab — CHANGES_REQUESTED** [review]
Please mention the return data type wherever the method returns.

File: src/app/shared/components/fy-duplicate-detection/fy-duplicate-detection.component.ts [hunk 4]
```diff
@@ -68,6 +62,12 @@ export class FyDuplicateDetectionComponent implements OnInit, ControlValueAccess
     }
   }
 
+  ngOnInit() {
```
Comment (rvab) [inline src/app/shared/components/fy-duplicate-detection/fy-duplicate-detection.component.ts:65]: Should it be ngOnInit or ionViewWillEnter?

**Review by rvab — APPROVED** [review]
There are still `get` methods with no return types. We should start mentioning it. Approving with an exception.

**Comment by yksolanki9** [comment]
Adding return types to methods across the codebase and updating component lifcycle methods will involve a lot of testing and modification which goes beyond the scope of this pr. 
I've added link to the comments and created a separate task, will fix these as a separate pr - https://app.clickup.com/t/2f88cgx

File: src/app/shared/components/fy-duplicate-detection/fy-duplicate-detection.component.ts (hunk 4 above)
Comment (yksolanki9) [reply src/app/shared/components/fy-duplicate-detection/fy-duplicate-detection.component.ts]: This file has been deleted in master

**Review by Julias0 — APPROVED** [review]

**Pull request merged**